tools/handle-ddl-lock.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Handling the Sharding DDL Lock Manually
+summary: Learn to manually unlock the sharding DDL lock.
+category: tools
+---
+
+# Handling the Sharding DDL Lock Manually
+
+This document shows how to manually unlock the sharding DDL lock in different cases.
+
+Generally, the sharding DDL lock synchronization of DM can be completed automatically. But when some abnormality happens, you need to use `unlock-ddl-lock`/`break-ddl-lock` to unlock the abnormal DDL lock manually.
+
+The possible causes of an abnormality include:
+
+- Some DM-workers go offline
+- Some DM-worker restarts (or is unreachable temporarily)
+- DM-master restarts
+
+> **Note:** You can use `unlock-ddl-lock`/`break-ddl-lock` only when you are definitely clear about the possible impacts brought by this command and you can accept the impacts.
+
+## Case one: some DM-workers go offline
+
+Before DM-master tries to automatically unlock the sharding DDL lock, all the DM-workers need to receive the sharding DDL event. If the sharding DDL operation is already in the synchronization process, and some DM-workers have gone offline and are not to be restarted, the sharding DDL lock cannot be automatically synchronized and unlocked since not all the DM-workers can receive the DDL event.
+
+If you do not need to make some DM-workers offline in the synchronization process of the sharding DDL operation, a better policy is to use `stop-task` to stop the running task first, then make some DM-workers offline, and finally use `start-task` and the **new task configuration** that does not contain the already offline DM-worker to restart the task.
+
+If the owner has finished the DDL operation but other DM-workers have not skipped this DDL operation, the owner has gone offline. For the solution, see [Case two: some DM-worker restarts](#case-two-some-dm-worker-restarts-or-is-unreachable-temporarily).
```

Comment:
This sentence does not quite make sense to me.  It would be helpful to highlight the cause/effect:
Is it:
"If the owner goes offline before the DDL operation is complete **then** the other DDL workers will block waiting..." ?
"If the owner goes offline before propagating XYZ state to other DM-workers **then** ..".?
"if the owner finishes the DDL operation but other DM-workers skip the DDL operation, then the owner will be forced into offline state" ?

Reply:
@csuzhangxc Please share your thoughts. Thanks.